backend/tests/services/test_external_context_sync.py

```
@@ -21,6 +22,23 @@
 from app.services.rag.sources import ExternalRefValidationError
 
 
+def test_invalid_external_ref_log_does_not_expose_resource_url(caplog) -> None:
+    secret_url = "https://example.test/document?token=secret"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "secret_url"

(S105)